openwhisk/executor.go

```diff
@@ -68,17 +87,151 @@ func NewExecutor(logout *os.File, logerr *os.File, command string, env map[strin
 	}
 	cmd.ExtraFiles = []*os.File{pipeIn}
 	output := bufio.NewReader(pipeOut)
-	return &Executor{
-		cmd,
-		input,
-		output,
-		make(chan bool),
+	e := &Executor{
+		cmd:    cmd,
+		input:  input,
+		output: output,
+		exited: make(chan bool),
+		logout: logout,
+		logerr: logerr,
+	}
+
+	// TODO: Where to surface configuration errors (unsupported destinations etc)? Ideally at action creation time.
+	if env["LOG_DESTINATION_SERVICE"] != "" {
+		if err := e.setupRemoteLogging(env); err != nil {
+			return nil
+		}
+	}
+
+	return e
+}
+
+func (proc *Executor) setupRemoteLogging(env map[string]string) error {
+	// A chunky buffer in order to not block execution of the function from dealing with log lines.
+	proc.lines = make(chan LogLine, 128)
+
+	// TODO: Should we throw stdout and stderr into one bucket? We'd lose the ability to
+	// distinguish them but, I think, we'd gain the ability to ensure that the ordering is
+	// equal to how the lines are ordered in-code (if it mixes stdout and stderr).
```

Comment:
I'm not quite parsing the question tbh. If we combine the two, we lose the ability to distinguish stdout/stderr. However, anecdotally, the AP implementation doesn't pass on any metadata about stdout/stderr either, so that might be okay? I guess it's not usual to log to stdout and stderr at the same time for apps and stuff. It's more common in CLI tools, where sometimes the rule is to write results to stdout and **any** logs (not just errors) to stderr.